app/Jobs/CreateQueryserviceBatchesJob.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace App\Jobs;
+
+use App\EventPageUpdate;
+use App\QsBatch;
+use App\QsCheckpoint;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Config;
+
+class CreateQueryserviceBatchesJob extends Job
+{
+    private const NAMESPACE_ITEM = 120;
+    private const NAMESPACE_PROPERTY = 122;
+    private const NAMESPACE_LEXEME = 146;
+
+    private int $entityLimit;
+
+    public function __construct()
+    {
+        $this->entityLimit = Config::get('wbstack.qs_batch_entity_limit');
+    }
+
+    public function handle(): void
+    {
+        DB::transaction(function () {
+            $latestCheckpoint = QsCheckpoint::get();
+
+            [$newEntities, $latestEventId] = $this->getNewEntities($latestCheckpoint);
+            foreach ($newEntities as $wikiId => $entityIdsFromEvents) {
+                try {
```

Comment:
Not sure about retrying vs. not failing all wikis on one naughty wiki here.

If we wanted the ideal behavior of:

- one naughty wiki does not block following wikis
- a failing wiki will be retried so we're sure this is not sth intermittent

we'd need to instead fan out each wiki into it's own job.

Not sure.

Reply:
I'm fine with this as it is; in general I would actually be "ok" with no try catch here because all of this work and logic is really just internal bookeeping inside the platform API.

Any failures here are not going to be due to some transient issue talking to a remote service over the network but are actually probably due to logical oversight in this one codebase.